draft-ietf-oauth-selective-disclosure-jwt.md

```diff
@@ -726,40 +764,48 @@ It is up to the Holder how to maintain the mapping between the Disclosures and t
- 3. If Key Binding is required, create a Key Binding JWT ((#kb-jwt)) and add it to the SD-JWT for presentation.
- 4. Send the SD-JWT for presentation to the Verifier.
+ 2. Assemble the SD-JWT, including the Issuer-signed JWT and the selected Disclosures (see (#data_formats) for the format).
+ 3. If Key Binding is not required:
+    1. Send the SD-JWT to the Verifier.
+ 4. If Key Binding is required:
+    1. Create a Key Binding JWT tied to the SD-JWT.
+    2. Assemble the SD-JWT+KB by concatenating the SD-JWT and the Key Binding JWT.
+    3. Send the SD-JWT+KB to the Verifier.
 
 ## Verification by the Verifier  {#verifier_verification}
 
-Upon receiving a Presentation, in addition to the checks outlined in (#sd_jwt_verification), Verifiers need to ensure that
+Upon receiving a presentation from a Holder, in the form of either an SD-JWT or
+an SD-JWT+KB, in addition to the checks outlined in (#sd_jwt_verification), Verifiers need to ensure that
 
- * if Key Binding is required, the Key Binding JWT is signed by the Holder and valid.
+ * if Key Binding is required, then the Holder has provided an SD-JWT+KB, and
+ * the Key Binding JWT is signed by the Holder and valid.
 
 To this end, Verifiers MUST follow the following steps (or equivalent):
 
 1. Determine if Key Binding is to be checked according to the Verifier's policy
    for the use case at hand. This decision MUST NOT be based on whether
    a Key Binding JWT is provided by the Holder or not. Refer to (#key_binding_security) for
    details.
-2. Process the SD-JWT as defined in (#sd_jwt_verification).
-3. If Key Binding is required:
-    1. If a Key Binding JWT is not provided, the Verifier MUST reject the Presentation.
-    2. Determine the public key for the Holder from the SD-JWT (see (#key_binding)).
-    3. Ensure that a signing algorithm was used that was deemed secure for the application. Refer to [@RFC8725], Sections 3.1 and 3.2 for details. The `none` algorithm MUST NOT be accepted.
-    4. Validate the signature over the Key Binding JWT per Section 5.2 of [@!RFC7515].
-    5. Check that the `typ` of the Key Binding JWT is `kb+jwt` (see (#kb-jwt)).
-    6. Check that the creation time of the Key Binding JWT, as determined by the `iat` claim, is within an acceptable window.
-    7. Determine that the Key Binding JWT is bound to the current transaction and was created for this Verifier (replay protection) by validating `nonce` and `aud` claims.
-    8. Calculate the digest over the Issuer-signed JWT and Disclosures as defined in (#integrity-protection-of-the-presentation) and verify that it matches the value of the `sd_hash` claim in the Key Binding JWT.
-    9. Check that the Key Binding JWT is a valid JWT in all other respects, per [@!RFC7519] and [@!RFC8725].
-
-If any step fails, the Presentation is not valid and processing MUST be aborted.
+2. If Key Binding is required and the Holder has provided an SD-JWT, the Verifier MUST reject the presentation.
```

Comment:
```suggestion
2. If Key Binding is required and the Holder has provided an SD-JWT without Key Binding, the Verifier MUST reject the presentation.
```

Reply:
I think the author of the PR would say that an SD-JWT is now without Key Binding by definition and that might suggest otherwise. How about this as a compromise on clarification? 
```suggestion
2. If Key Binding is required and the Holder has provided an SD-JWT (without Key Binding), the Verifier MUST reject the presentation.
```